rasa_core/training/structures.py

```diff
@@ -102,23 +113,129 @@ def add_event(self, event):
         if not self._is_action_listen(event):
             self.events.append(event)
 
+    @staticmethod
+    def _checkpoint_string(story_step_element):
+        return "> {}\n".format(story_step_element.as_story_string())
+
+    @staticmethod
+    def _user_string(story_step_element, prefix=''):
+        return "* {}{}\n".format(prefix, story_step_element.as_story_string())
+
+    def _store_user_strings(self, story_step_element, prefix=''):
+        self.as_story_string_helper.no_form_string += self._user_string(
+                story_step_element
+        )
+        self.as_story_string_helper.form_string += self._user_string(
+                story_step_element, prefix
+        )
+
+    @staticmethod
+    def _bot_string(story_step_element, prefix=''):
+        return "    - {}{}\n".format(prefix,
+                                     story_step_element.as_story_string())
+
+    def _store_bot_strings(self, story_step_element, prefix=''):
+        self.as_story_string_helper.no_form_string += self._bot_string(
+                story_step_element
+        )
+        self.as_story_string_helper.form_string += self._bot_string(
+                story_step_element, prefix
+        )
+
+    def _reset_stored_strings(self):
+        self.as_story_string_helper.form_string = ''
+        self.as_story_string_helper.no_form_string = ''
+
     def as_story_string(self, flat=False):
         # if the result should be flattened, we
         # will exclude the caption and any checkpoints.
+
+        for s in self.start_checkpoints:
+            if s.name == STORY_START:
+                # first story step in the story, so reset helper
+                self.as_story_string_helper = AsStoryStringHelper()
+
         if flat:
             result = ""
         else:
             result = "\n## {}\n".format(self.block_name)
             for s in self.start_checkpoints:
                 if s.name != STORY_START:
-                    result += "> {}\n".format(s.as_story_string())
+                    result += self._checkpoint_string(s)
+
         for s in self.events:
             if isinstance(s, UserUttered):
-                result += "* {}\n".format(s.as_story_string())
+                if self.as_story_string_helper.active_form is None:
```

Comment:
oh good 😅 but let's confirm that we really want this first before I go through this